### PR TITLE
tcl-tk: Fix KVO crash

### PR DESCRIPTION
**Pull request by chrstphrchvz**
This crash has been reported several times, particularly by Tkinter users (see e.g. python/cpython#92603); it will be fixed in Tk 8.6.14.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by chrstphrchvz** [comment]
I have opened https://github.com/orgs/Homebrew/discussions/4761 regarding the occasional CI failure when downloading the itk4 distfile from certain SourceForge mirrors. I noticed the same thing on #130971.

**Review by fxcoudert — APPROVED** [review]

File: Formula/t/tcl-tk.rb [hunk 1]
```diff
@@ -61,6 +61,13 @@ class TclTk < Formula
...
+
+    # Bugfix for KVO crash; will be in Tk 8.6.14
+    # See https://core.tcl-lang.org/tk/info/310c74ecf4
+    patch :p0 do
+      url "https://raw.githubusercontent.com/macports/macports-ports/6a93695d61d3/x11/tk/files/fix-kvo-crash.diff"
+      sha256 "ec9a9234b4a326e5621fe78e078c29aa4784b6dc88c59a43d828639ebae0af41"
+    end
   end
 
   resource "itk4" do
```
Comment (ZhongRuoyu) [inline Formula/t/tcl-tk.rb:73]: Not too sure about this one, but let's see if a mirror helps, like we do for the `tk` resource:

```
mirror "https://src.fedoraproject.org/lookaside/extras/itk/itk4.1.0.tar.gz/sha512/1deed09daf66ae1d0cc88550be13814edff650f3ef2ecb5ae8d28daf92e37550b0e46921eb161da8ccc3886aaf62a4a3087df0f13610839b7c2d6f4b39c9f07e/itk4.1.0.tar.gz"
```

Comment (chrstphrchvz) [reply Formula/t/tcl-tk.rb:73]: It seems the error encountered for the main URL is fatal and does not lead to retrying with the mirror.

Comment (ZhongRuoyu) [reply Formula/t/tcl-tk.rb:73]: Yea. But the bottles were built, so it's fine to merge this.

Still, would be nice to figure that out on `brew`'s side, though.

**Review by ZhongRuoyu — APPROVED** [review]

**Comment by ZhongRuoyu** [comment]
Thanks for the fix @chrstphrchvz! Merging now.

**Pull request merged**